### ISSUE TITLE
error: Argument 1 to "astype" of "DataFrame" has incompatible type "Series[Any]"  [arg-type]

### ISSUE THREAD

**Issue by randolf-scholz** [pandas-dev/pandas-stubs#410]
```python
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
df = df.astype(df.dtypes)  # ✘ raises [arg-type]
```

**Please complete the following information:**
- python 3.10.6
- mypy                          0.982
- pandas                        1.5.1
- pandas-stubs                  1.5.1.221024

**Comment by Dr-Irv** [comment]
Thanks for the report.  Need to add `Series` to `astype` argument in `core/frame.pyi` here:
https://github.com/pandas-dev/pandas-stubs/blob/001c0afb3ef0dfe21e96ced87c9e455c657a50c3/pandas-stubs/core/frame.pyi#L1362

PR with test welcome.

@randolf-scholz Note that we do not document that you can pass a `Series` to `DataFrame.astype()`, so can you create an issue in the pandas repo for that?

**Comment by gandhis1** [comment]
Can't you pass any `Iterable`, provided it has the same number of elements as the number of columns? And `Series` is just one example of that

**Comment by Dr-Irv** [comment]
closed via #413
